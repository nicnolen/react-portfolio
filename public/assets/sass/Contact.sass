@import './variables'
@import './index'
@import './App'

form
  // min-height: 85vh
  display: flex
  flex-direction: column
  align-items: center
  justify-content: center
  margin: 25px auto

form label 
  font-size: 1.2rem
  margin: 0.625rem 0 0.25rem
  color: $red

form label,
form input,
form textarea 
  display: block

input 
  font-size: 1rem
  padding: 0.25rem
  outline: none
  border-radius: 3px
  transition: box-shadow 0.2s
  border-radius: 10px

input:focus 
  border: 1px solid var(--secondary)
  box-shadow: 0 0 0.25rem rgba(0, 0, 0, 0.5)

textarea 
  width: 100%
  height: auto
  border: 3px solid #cccccc
  padding: 5px
  border-radius: 10px

button
  width: 37rem

.error 
  background-color: $gold
  color: $red
 