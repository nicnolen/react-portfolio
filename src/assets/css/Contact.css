/* PRIMARY COLORS */
/* BACKGROUND COLORS */
/* IMAGE FOLDER PATH */
/* FADE IMAGE */
/* PRIMARY COLORS */
/* BACKGROUND COLORS */
/* IMAGE FOLDER PATH */
/* FADE IMAGE */
/* Universal Selectors */
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
}

/* PRIMARY COLORS */
/* BACKGROUND COLORS */
/* IMAGE FOLDER PATH */
/* FADE IMAGE */
/* PRIMARY COLORS */
/* BACKGROUND COLORS */
/* IMAGE FOLDER PATH */
/* FADE IMAGE */
/* Universal Selectors */
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
}

.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #ffffff;
}

.App-link {
  color: #61dafb;
}

body {
  min-height: 100vh;
}

ul {
  list-style-type: none;
}

footer a, div a, li a, a {
  text-decoration: none;
}

.hero {
  background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url(../images/portfolioImages/bg-img.jpg);
  background-size: cover;
  background-position: center;
}

.top-title {
  color: #d49828;
  font-size: 6vh;
  text-align: center;
}

.subtitle {
  color: #9400d3;
  font-size: 4vh;
}

form {
  min-height: 82.9vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

form label {
  font-size: 1.2rem;
  margin: 0.625rem 0 0.25rem;
  color: #d2042d;
}

form label,
form input,
form textarea {
  display: block;
}

input {
  font-size: 1rem;
  padding: 0.25rem;
  outline: none;
  border-radius: 3px;
  transition: box-shadow 0.2s;
  border-radius: 10px;
}

input:focus {
  border: 1px solid var(--secondary);
  box-shadow: 0 0 0.25rem rgba(0, 0, 0, 0.5);
}

textarea {
  width: 600px;
  height: 120px;
  border: 3px solid #cccccc;
  padding: 5px;
  border-radius: 10px;
}

button {
  width: 37rem;
}

/*# sourceMappingURL=Contact.css.map */
