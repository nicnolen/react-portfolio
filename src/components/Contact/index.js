import { send } from 'emailjs-com';
import React, { useState } from 'react';
import { validateEmail } from '../../utils/helpers';
import '../../assets/css/Contact.css';

function ContactForm() {
  const [toSend, setToSend] = useState({
    from_name: '',
    to_name: '',
    message: '',
    reply_to: '',
  });
  const [errorMessage, setErrorMessage] = useState('');
  const { from_name, to_name, message, reply_to } = toSend;

  // Handle change function
  const handleChange = e => {
    // validate email
    if (e.target.name === 'email') {
      const isValid = validateEmail(e.target.value);
      // isValid conditional statement
      if (!isValid) {
        setErrorMessage('Invalid email format!');
      } else {
        setErrorMessage('');
      }
    } else {
      // error message for blank elements
      if (!e.target.value.length) {
        setErrorMessage(`${e.target.name} is required!`);
      } else {
        setErrorMessage('');
      }
    }
    // if there are no error messages
    if (!errorMessage) {
      // update the formState value for the name property
      setToSend({ ...toSend, [e.target.name]: e.target.value });
    }
  };

  // Handle submit function
  const handleSubmit = e => {
    e.preventDefault();
    send('service_k06hlfp', 'template_iq5w2c2', toSend, 'Rau85KwJyF-kPthpC')
      .then(response => {
        console.info('SUCCESS!', response.status, response.text);
      })
      .catch(err => {
        console.error('FAILED...', err);
      });
  };

  return (
    <section>
      <form onSubmit={handleSubmit}>
        <div>
          <label htmlFor="from_name" className="pr-2">From:</label>
          <input
            type="text"
            name="from_name"
            value={from_name}
            onChange={handleChange}
            className="mb-2"
          />
        </div>
        <div>
          <label htmlFor="to_name">To:</label>
          <input
            type="text"
            name="to_name"
            value={to_name}
            onChange={handleChange}
            className="mb-2"
          />
        </div>
        <div>
          <label htmlFor="email">Reply To:</label>
          <input
            type="text"
            name="reply_to"
            value={reply_to}
            onChange={handleChange}
            className="mb-2"
          />
        </div>
        <div>
          <label htmlFor="message">Message:</label>
          <textarea
            name="message"
            rows="5"
            value={message}
            onChange={handleChange}
            className="mb-2"
          />
          {errorMessage && (
            <div>
              <p className="errorText">{errorMessage}</p>
            </div>
          )}
        </div>
        <button type="submit" className="btn btn-info">Submit</button>
      </form>
    </section>
  );
}

export default ContactForm;
