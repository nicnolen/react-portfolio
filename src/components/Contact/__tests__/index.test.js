// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('Contact component', () => {
  // verify the component is rendering properly
  it('Contact component renders', () => {
    render(<Contact />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });
});
