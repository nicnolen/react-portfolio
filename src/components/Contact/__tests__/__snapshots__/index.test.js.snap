// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="container"
    style="max-width: 660px;"
  >
    <h2
      class="top-title mt-5 mb-3"
    >
      Contact Me
    </h2>
    <form>
      <div
        style="width: 75%; margin: 0px auto;"
      >
        <label
          for="to_name"
        >
          To:
        </label>
        <input
          class="mb-2"
          name="to_name"
          type="text"
          value="nicnolen@ymail.com"
        />
      </div>
      <div
        style="width: 75%; margin: 0px auto;"
      >
        <label
          class="pr-2"
          for="from_name"
        >
          From:
        </label>
        <input
          class="mb-2"
          name="from_name"
          type="text"
          value=""
        />
      </div>
      <div
        style="width: 75%; margin: 0px auto;"
      >
        <label
          for="name"
        >
          Name:
        </label>
        <input
          class="mb-2"
          name="name"
          type="text"
          value=""
        />
      </div>
      <div
        style="width: 75%; margin: 0px auto;"
      >
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          class="mb-2"
          name="message"
          rows="5"
        />
      </div>
      <button
        class="btn btn-info mb-5"
        style="width: 75%; margin: 0px auto;"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
